src/smt/smt_solver.cpp

```diff
@@ -66,6 +66,16 @@ void SmtSolver::finishInit(const LogicInfo& logicInfo)
     theory::TheoryConstructor::addTheory(d_theoryEngine.get(), id);
   }
 
+  // Add the proof checkers for each theory
+  if (d_pnm)
+  {
+    ProofChecker* pc = d_pnm->getChecker();
+    for (theory::TheoryId id = theory::THEORY_FIRST; id < theory::THEORY_LAST;
```

Comment:
I'd prefer this loop was inside of TheoryEngine::initializeProofChecker, meaning TheoryEngine::initializeProofChecker should not take a TheoryId.

Reply:
I don't think there is a reason to be this fine-grained.  I think it'd best to keep it simple, e.g. say a special preprocessing pass used an arithmetic rule.

Reply:
Ah, so you mean we should always just initialize them all?

Reply:
Yes

Reply:
I see. I'll update.